openstudiocore/src/gbxml/ReverseTranslator.cpp

```diff
@@ -725,26 +732,95 @@ namespace gbxml {
       }
 
       if (space && adjacentSpaceElements.size() == 2){
-
         QString adjacentSpaceId = adjacentSpaceElements.at(1).toElement().attribute("spaceIdRef");
         auto adjacentSpaceIt = m_idToObjectMap.find(adjacentSpaceId);
-        if (adjacentSpaceIt != m_idToObjectMap.end()){
+        if (adjacentSpaceIt != m_idToObjectMap.end()) {
           boost::optional<model::Space> adjacentSpace = adjacentSpaceIt->second.optionalCast<openstudio::model::Space>();
-          if (adjacentSpace){
-            // DLM: we have issues if interior ceilings/floors are mislabeled, override surface type for adjacent surfaces
-            // http://code.google.com/p/cbecc/issues/detail?id=471
+          if (adjacentSpace) {
+
             std::string currentSurfaceType = surface.surfaceType();
-            surface.assignDefaultSurfaceType();
-            if (currentSurfaceType != surface.surfaceType()){
-              LOG(Warn, "Changing surface type from '" << currentSurfaceType << "' to '" << surface.surfaceType() << "' for surface '" << surface.name().get() << "'");
+            if (currentSurfaceType == "RoofCeiling" || currentSurfaceType == "Floor") {
+              // DLM: we have issues if interior ceilings/floors are mislabeled, override surface type for adjacent surfaces
+              // http://code.google.com/p/cbecc/issues/detail?id=471
+              // Try to figure it out based on the surfaceType
+              bool figuredOut = false;
+              QString spaceSurfaceType = adjacentSpaceElements.at(0).toElement().attribute("surfaceType");
+              QString adjacentSpaceSurfaceType = adjacentSpaceElements.at(1).toElement().attribute("surfaceType");
+              if (!spaceSurfaceType.isEmpty()) {
+                if (currentSurfaceType == "Floor") {
+                  if (spaceSurfaceType == "InteriorFloor") {
+                    // No changes
+                    figuredOut = true;
+                  } else if (spaceSurfaceType == "Ceiling") {
+                    // Swap roles of space and adjacentSpace
+                    surface.setSpace(*adjacentSpace);
+                    auto temp = space;
+                    space = adjacentSpace;
+                    adjacentSpace = temp;
+                    figuredOut = true;
+                  }
+                } else if (currentSurfaceType == "RoofCeiling") {
+                  if (spaceSurfaceType == "Ceiling") {
+                    // No changes
+                    figuredOut = true;
+                  } else if (spaceSurfaceType == "InteriorFloor") {
+                    // Swap roles of space and adjacentSpace
+                    surface.setSpace(*adjacentSpace);
+                    auto temp = space;
+                    space = adjacentSpace;
+                    adjacentSpace = temp;
+                    figuredOut = true;
+                  }
+                }
+              } else if (!adjacentSpaceSurfaceType.isEmpty()) {
```

Comment:
@jasondegraw is this block ever going to be hit?  the first attribute is empty but the second is set?  if so, should this logic be reversed?  e.g. the current surface is a floor which is ok if the adjacent surface is a ceiling?

Reply:
@macumber I'm not sure why this is showing here as a diff, because it's in develop already. In any case, it could be hit if the surface type isn't specified for both spaces, and I think we've agreed we'd prefer to see that specified for both spaces. I think you're right, the logic needs to be reversed (due the space role swap if currentSurfaceType and adjacentSurfaceType agree).